_posts/2023-06-27-redis.md

```diff
@@ -0,0 +1,23 @@
+---
+layout: post
+title: "Redis"
+description: "Why is redis fast, what are the pros and cons of using redis"
+comments: true
+keywords: "redis, infrastructure"
+published: false
+---
+
+[Redis](https://redis.io) was designed to be a blazingly fast key-value store to make caching easy. And for most parts, it lives up to the billing. Setting up and using Redis has always been a breeze. The success and popularity of Redis has made it a very powerful tool in the software engineers' arsenal; with its support of almost all data types and associated operations out of the box, it is a perfect solution for almost all persistence problems. As with everything, people have used it in ways without considering its fitness for the problem at hand, often to great success.
```

Comment:
Not to be pendantic but this is an important clarificaiton. I think if you listen to Antirez, he appears to not have written redis to cache data per se. His writings about why he wrote it implies he wrote redis because he wanted a fast in-memory data structure server so the more correct way to describe redis is `a key-value in-memory database that trades some durability for speed`. That it's used commonly for caching is a side-effect of its speed.

You can see clearly how redis has progressed over time confirms that. It's rich in data structures from common ones to esoteric ones like hyperloglog and bloom filters. It even has full blown distributed locking support and data streaming primitives now.